vcd/resource_vcd_vapp_network.go

```diff
@@ -271,17 +448,92 @@ func resourceVappNetworkDelete(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return fmt.Errorf("error removing vApp network: %#v", err)
 	}
 
-	err = task.WaitTaskCompletion()
+	d.SetId("")
+
+	return nil
+}
+
+// resourceVcdVappNetworkImport is responsible for importing the resource.
+// The following steps happen as part of import
+// 1. The user supplies `terraform import _resource_name_ _the_id_string_` command
+// 2. `_the_id_string_` contains a dot formatted path to resource as in the example below
+// 3. The functions splits the dot-formatted path and tries to lookup the object
+// 4. If the lookup succeeds it sets the ID field for `_resource_name_` resource in statefile
+// (the resource must be already defined in .tf config otherwise `terraform import` will complain)
+// 5. `terraform refresh` is being implicitly launched. The Read method looks up all other fields
+// based on the known ID of object.
+//
+// Example resource name (_resource_name_): vcd_vapp_network.network_name
+// Example import path (_the_id_string_): org-name.vdc-name.vapp-name.network-name
+func resourceVcdVappNetworkImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("[vApp network import] resource name must be specified as org-name.vdc-name.vapp-name.network-name")
+	}
+	orgName, vdcName, vappName, networkName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
 	if err != nil {
-		return fmt.Errorf("error waiting for task to complete: %+v", err)
+		return nil, fmt.Errorf("[VM import] unable to find VDC %s: %s ", vdcName, err)
 	}
 
-	d.SetId("")
+	vapp, err := vdc.GetVAppByName(vappName, false)
+	if err != nil {
+		return nil, fmt.Errorf("[VM import] error retrieving vapp %s: %s", vappName, err)
+	}
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("[VM import] error retrieving vApp network configuration %s: %s", networkName, err)
+	}
 
-	return nil
+	vappNetworkToImport := types.VAppNetworkConfiguration{}
+	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == networkName {
+			vappNetworkToImport = networkConfig
+			break
+		}
+	}
+
+	if vappNetworkToImport == (types.VAppNetworkConfiguration{}) {
+		return nil, fmt.Errorf("didn't find vApp network: %s", networkName)
+	}
+
+	networkId, err := govcd.GetUuidFromHref(vappNetworkToImport.Link.HREF)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get network ID from HREF: %s", err)
+	}
+
+	d.SetId(networkId)
+
+	if vcdClient.Org != orgName {
+		_ = d.Set("org", orgName)
+	}
+	if vcdClient.Vdc != vdcName {
+		_ = d.Set("vdc", vdcName)
+	}
+	_ = d.Set("name", networkName)
+	_ = d.Set("vapp_name", vappName)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceVcdDhcpPoolHash(v interface{}) int {
```

Comment:
How does it differ from default function. I believe I read somewhere in the code that the best option is to not use custom Hash functions (if possible)

Reply:
I guess not much, I just improved already existing hash function which wasn't working in one case. Changing to default now requires to messup with scheme due SDK function requires scheme part to create hash and retesting.